### PR TITLE
PoS Mempool

### PR DESCRIPTION
**Pull request by AeonSw4n**

### CODE REVIEW

**Comment by mattfoley8** [comment]
This should use feature/proof-of-stake as its parent branch.

File: lib/mempool_test.go [hunk 1]
```diff
@@ -436,3 +439,60 @@ func TestMempoolAugmentedUtxoViewTransactionChain(t *testing.T) {
 		mp.Stop()
 	})
 }
+
+func TestFeeTimeBucket(t *testing.T) {
+	seed := time.Now().UnixNano()
```
Comment (mattfoley8) [inline lib/mempool_test.go:444]: Non-determinism in CI is bad. We should use a static seed.

File: lib/block_producer.go [hunk 2]
```diff
@@ -494,6 +494,106 @@ func (desoBlockProducer *DeSoBlockProducer) UpdateLatestBlockTemplate() error {
...
+	//  - Then, we will adjust the block reward to the appropriate amount.
+	totalFeeNanos := uint64(0)
+	feesUtxoView, err := NewUtxoView(desoBlockProducer.chain.db, desoBlockProducer.params,
+		desoBlockProducer.postgres, desoBlockProducer.chain.snapshot)
+	if err != nil {
+		return nil, err
+	}
+	// Skip the block reward, which is the first txn in the block.
+	for _, txnInBlock := range blockRet.Txns[1:] {
+		var feeNanos uint64
```
Comment (diamondhands0) [inline lib/block_producer.go:568]: FIXME: Need to react to StateConnected bool here.

File: lib/block_producer.go [hunk 3]
```diff
@@ -494,6 +494,106 @@ func (desoBlockProducer *DeSoBlockProducer) UpdateLatestBlockTemplate() error {
...
+		}
+
+		// Add the fee to the block reward output as we go. Note this has some risk of
+		// increasing the size of the block by one byte, but it seems like this is an
+		// extreme edge case that goes away as soon as the function is called again.
+		totalFeeNanos += feeNanos
+	}
+
+	// TODO: BMF stuff
+	blockRewardOutput.AmountNanos = CalcBlockRewardNanos(uint32(blockRet.Header.Height)) + totalFeeNanos
```
Comment (diamondhands0) [inline lib/block_producer.go:583]: FIXME: BMF is on a per-txn basis.

**Comment by AeonSw4n** [comment]
Closed in favor of #561 and #562

**Pull request closed without merging**